src/guiapi/src/IWindowMenuItem.cpp

```diff
@@ -86,3 +83,17 @@ Rect16 IWindowMenuItem::getRollingRect(IWindowMenu &window_menu, Rect16 rect) co
     rect -= irc.Width();
     return rect;
 }
+
+// Returns custom width Rectangle, aligned intersection on the right of the base_rect
+Rect16 IWindowMenuItem::getCustomRect(IWindowMenu &window_menu, Rect16 base_rect, uint16_t custom_rect_width) {
+    Rect16 custom_rect = { base_rect.Left(), base_rect.Top(), custom_rect_width, base_rect.Height() };
+    custom_rect += Rect16::Left_t(base_rect.Width() - custom_rect.Width());
```

Comment:
this could be possibly rewritten into one expression, which will improve the chance of optimizing the whole function